File: dotnet/Pyrolite/Pickle/IObjectPickler.cs
/* part of Pyrolite, by Irmen de Jong (irmen@razorvine.net) */

using System.IO;

namespace Razorvine.Pickle
{
	
/// <summary>
/// Interface for object Picklers used by the pickler, to pickle custom classes. 
/// </summary>
public interface IObjectPickler {
	/**
	 * Pickle an object.
	 */
	void pickle(object o, Stream outs, Pickler currentPickler);
}

}

File: java/src/main/java/net/razorvine/pickle/IObjectPickler.java
package net.razorvine.pickle;

import java.io.IOException;
import java.io.OutputStream;

/**
 * Interface for Object Picklers used by the pickler, to pickle custom classes. 
 *
 * @author Irmen de Jong (irmen@razorvine.net)
 */
public interface IObjectPickler {
	/**
	 * Pickle an object.
	 */
	public void pickle(Object o, OutputStream out, Pickler currentPickler) throws PickleException, IOException;
}
